src/main/resources/avro/referencemethods.avdl

```diff
@@ -325,4 +396,99 @@ GetSequenceBasesResponse getSequenceBases(
     /** Additional request parameters to restrict the query. */
     GetSequenceBasesRequest request) throws GAException;
 
+/****************  /joins/search  *******************/
+/**
+This request maps to the body of `POST /joins/search`
+as JSON.
+*/
+record SearchJoinsRequest {
+  /**
+  If present, return only `Join`s which belong to this reference set.
+  */
+  union { null, string } referenceSetId = null;
+  
+  /**
+  If present, return only `Join`s which belong to this variant set.
+  */
+  union { null, string } variantSetId = null;
+
+  // TODO: Specify a proper sort of filter object instead of this. We want to be
+  // able to support a list of sequences, but also a single sequence and a
+  // range.
+
+  /**
+  If nonempty, return only `Join`s with at least one `Position` on one of these
+  sequences.
+  */
+  array<string> sequenceIds = [];
+  
+  /**
+  Must not contain more entries than `sequenceIds`.
```

Comment:
Jerome, yes it is intended to be flexible, since it's just a definition as Cassie (@cassiedoll) mentioned a while ago in the following post https://github.com/ga4gh/schemas/issues/131#issuecomment-53291001.  So it's up to the implementation to decide the most reasonable, elegant and efficient algorithms to provide the functionality, which means you have almost complete freedom :)

~p